pkg/algorithm/nodeselect/v1alpha1/select_node.go

```diff
@@ -187,6 +210,96 @@ func (ac *Config) getBlockDevice() (*ndmapis.BlockDeviceList, error) {
 	if err != nil {
 		return nil, err
 	}
-	bdl := bdL.Filter(blockdevice.FilterClaimedDevices, blockdevice.FilterNonInactive)
+	bdl := bdL.Filter(blockdevice.FilterNonInactive)
 	return bdl.BlockDeviceList, nil
 }
+
+//TODO: Make changes in below code befor PR checked in
+// 1) Use Builder Pattern or some approach to present below code
+// 2) Refactor the below code before removing WIP
+
+// ClaimBlockDevice will create BDC for corresponding BD
+func (ac *Config) ClaimBlockDevice(nodeBDs *nodeBlockDevice, spc *apis.StoragePoolClaim) (*NodeClaimedBDDetails, error) {
+	nodeClaimedBDs := &NodeClaimedBDDetails{
+		NodeName:        "",
+		BlockDeviceList: []ClaimedBDDetails{},
+	}
+
+	namespace := env.Get(env.OpenEBSNamespace)
+	bdcKubeclient := bdc.NewKubeClient().
+		WithNamespace(namespace)
+	labels := map[string]string{string(apis.StoragePoolClaimCPK): spc.Name}
+	nodeClaimedBDs.NodeName = nodeBDs.NodeName
+
+	for _, bdName := range nodeBDs.BlockDevices.Items {
+		var hostName, bdcName string
+		claimedBD := ClaimedBDDetails{}
+
+		bdObj, err := ac.BlockDeviceClient.Get(bdName, metav1.GetOptions{})
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get block device {%s}", bdName)
+		}
+		hostName = bdObj.Labels[string(apis.HostNameCPK)]
+
+		if bdObj.IsClaimed() {
+			//TODO: Use HasLabel Predicate for below check
+			bdcName = bdObj.Spec.ClaimRef.Name
+			bdcObj, err := bdcKubeclient.Get(bdcName, metav1.GetOptions{})
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to get block device claim{%s}", bdcName)
+			}
+			if bdcObj.Labels[string(apis.StoragePoolClaimCPK)] != spc.Name {
+				return nil, errors.Errorf("block device %s is already in use", bdName)
+			}
+		} else {
+			bdcName = "bdc-" + string(bdObj.UID)
```

Comment:
If not Claimed, check whether there is already pending BDC asking for this BD

Reply:
- Get the list of BDCs related to this SPC
- Check if any BDC of it is trying to claim this BD
- If so, go to next BD in the list, else, create BDC for this BD

Reply:
Done